File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int x: stones){
            pq.add(x);
        }
        System.out.println(pq);
        while(pq.size() > 1){
            int a = pq.poll();
            int b = pq.poll();
            System.out.println(pq);
            if(a-b != 0){
                pq.add(Math.abs(a-b));
            }
        }
        return pq.isEmpty() ? 0 : pq.poll();
    }
}

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones = [-i for i in stones]
        heapq.heapify(stones)
        while stones:
            if len(stones) == 1:
                return -stones[0]
            x, y = heapq.heappop(stones), heapq.heappop(stones)
            if x-y != 0:
                heapq.heappush(stones, x-y)
        return 0
        